carbon/components/OptionsSwitcher/index.tsx

```diff
@@ -1,37 +1,78 @@
+"use client";
 /**
  * A widget to choose betwwen different options
+ * name: name of the widget (for state persistence state)
+ * options: Options offered by the widget
+ * onSelectionChange: function triggered when an option is selected
+ * value: initial value of the widget
+ * className: className to apply to the widget
  */
 import { OptionChangeHandler, Options } from "lib/charts/options";
-import { Key, useState } from "react";
+import { Key, useEffect, useState } from "react";
 import styles from "./styles.module.scss";
 
 export default function OptionsSwitcher<T extends Key>(props: {
+  name: string;
   options: Options<T>;
-  onSelectionChange: OptionChangeHandler<T>;
+  onSelectionChange?: OptionChangeHandler<T>;
   value?: T;
   className?: string;
+  readonly?: boolean;
 }) {
+  // The widget initial value is the first option value or the value from the url query
+  let initialValue = props.value || props.options[0].value;
+  if ("undefined" !== typeof window) {
+    const url = new URL(window.location.href);
+    initialValue = (url.searchParams.get(props.name) as T) || initialValue;
+  }
   const [selected, setSelected] = useState<T>(
     props.value || props.options[0].value
   );
+  const readonly = props.readonly === undefined ? false : props.readonly;
```

Comment:
There's an [operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) for that! 😄 
```suggestion
  const readonly = props.readonly ?? false;
```

Reply:
And you showed it to me already but i am a slow learner :)